__tests__/app.test.js

```diff
@@ -161,4 +161,41 @@ describe("/api/articles/:article_id/comments",()=>{
                 expect(data.body.msg).toBe("Invalid ID")
             })
         })
+    })
+
+    describe("7. POST /api/articles/:article_id/comments",()=>{
+        const commentData = {
+            author: "icellusedkars",
+            body: "dam, did you see what heskey did",
+            article_id: 1
+        }
+        test("post valid comment should return with 200, the comment with a unique id",()=>{
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send(commentData)
+            .expect(200)
+            .then(data=>{
+                expect(data.body.comment).toEqual(expect.objectContaining({
+                    body: "dam, did you see what heskey did",
+                    article_id: 1,
+                    author: "icellusedkars",
+                    comment_id: 19,
+                    created_at: expect.any(String)
+                }))
+            })
+        })
+        test("POST /api/articles/:article_id/comments when username does not exist",()=>{
+            const commentData = {
+                author: "heskey",
+                body: "dam, did you see what heskey did",
+                article_id: 1
+            }
+            return request(app)
+            .post("/api/articles/1/comments")
+            .send(commentData)
+            .expect(404)
+            .then((data)=>{
+                expect(data.body.msg).toBe("Username does not exist")
+            })
+        })
     })
```

Comment:
I would remove the article_id from the object here, you are already providing it in the url and as such you can take it from the url in your controller from req.params

Reply:
I think theres a few more tests here to do as well;
Posting to an article number that doesnt exist
posting to an article with an id that isnt even a number
posting a comment that doesnt even have a username
posting a comment that has a valid username but not a body?